server/src/server.js

```diff
@@ -576,7 +579,7 @@ app.post("/api/login", function(req, res) {
   });
 });
 
-app.put("/data/:id/:domain", function(req, res) {
+app.put("/data/:id/:domain", upload.single('blob'), function(req, res) {
```

Comment:
This doesn't appear to work with our test script, but I'm guessing also doesn't work with an old version of the add-on (which we'll need to support).

It's possible this could use a different endpoint, or maybe there's a way to use the upload functionality not as middleware.